praxis-isncsci/ui
# [Cell input] Update the domain model to support the selection of multiple cells

Closes #123 

## Problem

The current implementation only supports the selection of just one cell. We need to have an active cell and be able to select multiple other cells.

## Solution

We are removing the `selectPoint.useCase` and provider methods. We are introducing the `setActiveCell.useCase`. It contains the logic to change the selection of the active cell which affects the selected cells. We are supporting selection modes for single cell, multiple cells, and cell ranges.

## Testing

- [x] 1. All unit tests pass

---

- [x] 2. Can select a cell by clicking on it

<details>
  <summary>Test case</summary>

  1. `npm run build`
  2. `npm start`
  3. Navigate to **Storybook** story `http://localhost:6006/?path=/story/app-app--primary`
  4. On the grid, click on a cell
  5. Confirm that it is highlighted with a purple border
  6. Click on a different cell
  7. Confirm the previous selection gets cleared and the new cell gets highlighted

</details>
